messaging/messaging.cc

```diff
@@ -65,7 +65,9 @@ SubSocket * SubSocket::create(Context * context, std::string endpoint, std::stri
     return s;
   } else {
     delete s;
-    return NULL;
+    throw std::runtime_error(
+      "Failed to connect to " + endpoint + ": " + std::string(strerror(errno))
+    );
```

Comment:
we tend to avoid cpp exceptions. you can print to stderr instead

Reply:
Ok, i guess that works too